app/models/hardware.rb

```diff
@@ -44,6 +43,11 @@ def mac_addresses
   def ram_size_in_bytes
     memory_mb.to_i * 1.megabyte
   end
+  # resulting sql: "(CAST("hardwares"."memory_mb" AS bigint) * 1048576)"
+  virtual_attribute :ram_size_in_bytes, :integer, :arel => (lambda do |t|
+    t.grouping(Arel::Nodes::Multiplication.new(
+      Arel::Nodes::NamedFunction.new("CAST", [t[:memory_mb].as("bigint")]), 1.megabyte))
+  end)
```

Comment:
Arel... 😱 .... @kbrock @NickLaMuro Can you review this?  I have no idea.

@yrudman Can you add a test that recreates the problem? I don't understand how it overflows `ram_size_in_bytes` for a single vm in a report.  I was under the impression it was a problem aggregating many hardwares for many vm/instances that caused the overflow.